### PR TITLE
Freshen up examples/ directory

### PR DESCRIPTION
**Pull request by paulromano**
This PR makes a variety of changes in the `examples` directory:

- I've removed the `xml` subdirectory and moved all the `python` examples down one level. Since we don't encourage users to write XML files directly, this is in keeping line with that and encouraging them to go straight to Python. Right now, I haven't included the generated XML files for each of the examples, but I would be open to doing so.
- The `simple` example has been removed and I've replaced it with a "real" model that is truly simple -- Jezebel, which is a critical sphere of plutonium. That example demonstrates the very basics of creating a model.
- All the pincell models have been improved, making sure the use of the Python API is idiomatic. I also added a flux spectrum tally on the pincell example and an associated script to plot it.
- I've changed the `custom_source` example a little bit, putting the ring source in a box and adding a script that plots a flux mesh tally to demonstrate that the custom source works.
- I've added an `assembly` example.

### CODE REVIEW

File: examples/assembly/assembly.py [hunk 1]
```diff
@@ -0,0 +1,140 @@
...
+    clad_cell = openmc.Cell(fill=clad, region=+fuel_or & -clad_or)
+    hot_water_cell = openmc.Cell(fill=hot_water, region=+clad_or)
+
+    univ = openmc.Universe(name='Fuel Pin')
+    univ.add_cells([fuel_cell, clad_cell, hot_water_cell])
+    return univ
+
+
+def guide_tube_pin():
+    """Returns a control rode guide tube pin universe"""
```
Comment (pshriwise) [inline examples/assembly/assembly.py]:
```suggestion
    """Returns a control rod guide tube universe."""
```

File: examples/assembly/assembly.py [hunk 2]
```diff
@@ -0,0 +1,140 @@
...
+    assembly.universes[gt_pos[:, 0], gt_pos[:, 1]] = guide_tube_pin()
+
+    # Create outer boundary of the geometry to surround the lattice
+    outer_boundary = openmc.model.rectangular_prism(
+        pitch, pitch, boundary_type='reflective')
+
+    # Create a cell filled with the lattice
+    main_cell = openmc.Cell(fill=assembly, region=outer_boundary)
+
+    # Finally, create geometry by giving a list of cells filling the root
```
Comment (pshriwise) [inline examples/assembly/assembly.py]:
```suggestion
    # Finally, create geometry by providing a list of cells that fill the root
```

Comment (pshriwise) [reply examples/assembly/assembly.py]: Minor change. Take or leave :)

File: examples/custom_source/show_flux.py [hunk 3]
```diff
@@ -0,0 +1,18 @@
...
+# Plot the flux
+fig, ax = plt.subplots()
+ax.imshow(flux, origin='lower', extent=(-5.0, 5.0, -5.0, 5.0))
+ax.set_xlabel('x [cm]')
+ax.set_ylabel('y [cm]')
+plt.show()
+
+# If all worked well, you should see a ring "imprint" as well as a higher flux
+# to the right side (since the custom source has all particles moving in the
+# positive x direction))
```
Comment (pshriwise) [inline examples/custom_source/show_flux.py]:
```suggestion
# positive x direction)
```

**Comment by paulromano** [comment]
Thanks for the review @pshriwise

File: examples/custom_source/show_flux.py [hunk 4]
```diff
@@ -0,0 +1,18 @@
...
+    flux.shape = (50, 50)
+
+# Plot the flux
+fig, ax = plt.subplots()
+ax.imshow(flux, origin='lower', extent=(-5.0, 5.0, -5.0, 5.0))
+ax.set_xlabel('x [cm]')
+ax.set_ylabel('y [cm]')
+plt.show()
+
+# If all worked well, you should see a ring "imprint" as well as a higher flux
```
[the remark this answers is not in the document]
Comment (paulromano) [reply examples/custom_source/show_flux.py]: This script doesn't write an image file -- it shows the image interactively (`plt.show()`).

[the remark this answers is not in the document]
Comment (pshriwise) [reply examples/custom_source/show_flux.py]: Oh, I meant that the user may never open `show_flux.py` and see the comment here explaining what they should expect to see. In fact, now that there's a README (thank you!!), this comment would fit nicely there with a quick instruction about running `show_flux.py` when OpenMC is complete.

[the remark this answers is not in the document]
Comment (paulromano) [reply examples/custom_source/show_flux.py]: Oh indeed. good call

**Comment by pshriwise** [comment]
Thanks @paulromano!

**Pull request merged**